app/components/Swapbox.tsx

```diff
@@ -2,11 +2,117 @@
 
 import { Button, IconButton } from "swapr-ui";
 import { SwapInput } from "./ui/SwapInput";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { erc20Abi, formatEther, parseEther, Address } from "viem";
+import { useAccount, useConfig, useReadContract } from "wagmi";
+import { waitForTransactionReceipt, writeContract } from "wagmi/actions";
+import MarketABI from "@/abi/market.json";
+
+const WXDAI_ADDRESS = "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d";
+
+// Removes the given fraction from the given integer-bounded amount and returns the value as an original type.
+const removeFraction = (amount: bigint, fraction: number): bigint => {
+  if (fraction >= 1 || fraction <= 0)
+    throw `The given basisPoints ${fraction} is not in the range [0, 1].`;
+
+  const basisPoints = 10000;
+
+  const fractionInBasisPoints = fraction * basisPoints;
+
+  const keepFraction = basisPoints - fractionInBasisPoints;
+
+  return (amount * BigInt(keepFraction)) / BigInt(basisPoints);
+};
+
+export const Swapbox = ({ id }: { id: Address }) => {
+  const { address } = useAccount();
+  const config = useConfig();
 
-export const Swapbox = () => {
   const [tokenInAmount, setTokenInAmount] = useState("");
   const [tokenOutAmount, setTokenOutAmount] = useState("");
+  const [minAmountOut, setMinAmountOut] = useState<bigint>();
+
+  const outcomeIndex = 0;
+  const slippage = 0.01;
+
+  const amountWei = parseEther(tokenInAmount);
+
+  const { data: buyAmount } = useReadContract({
+    abi: MarketABI,
+    address: id,
+    functionName: "calcBuyAmount",
+    args: [amountWei, outcomeIndex],
+    query: { enabled: !!tokenInAmount },
+  });
+
```

Comment:
just an idea to explor, if we want to extract this into a file,

this could be on eg. `models/market` and be a function like.

`getBuyAmount({id, tokenInAmount})`

`amountWei` could be calculated inside..

Same as the ones bellow.

Reply:
I like to have this logic next to where it's being used. But I think we can abstract some things, like creating a  `useReadMarketContract` hook. If needed we can even abstract more things in the future.